exporter/prometheusexporter/README.md

```diff
@@ -19,6 +19,11 @@ The following settings can be optionally configured:
 - `metric_expiration` (default = `5m`): defines how long metrics are exposed without updates
 - `resource_to_telemetry_conversion`
   - `enabled` (default = false): If `enabled` is `true`, all the resource attributes will be converted to metric labels by default.
+- `prometheus_convention` (default = false): If `true`, the OpenTelemetry
+  metric names are transformed to follow [Prometheus metrics naming
+  conventions](https://prometheus.io/docs/practices/naming/)
+- `rename`: A map to rename metrics before exporting them (takes
```

Comment:
I understand your concerns.

The problem is that processors run **before** exporters in the pipeline. The idea here is to be able to override this automatic conversion with a manually-defined mapping, for a few metrics. This is just a workaround mechanism to allow the user to cover edge cases (when one Otel metric is badly defined, etc.) 

Instead of `rename`, we could have a `disable_automatic_renaming` option where the user could list the metrics that should be left unchanged (the necessary renaming would have been taken care of in **metrictransformprocessor**). But it's making things more complicated for the end user, and code wouldn't be much simpler.

Reply:
> The problem is that processors run before exporters in the pipeline.

@bertysentry I am not sure if it add that much more complexity. Just one more processor to refer to. And most users probably will use single exporter anyway.

If you want to have e.g. two exporters and make changes only for one of them, then actually [two pipelines referring the same receiver](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/design.md#receivers) can be created (so e.g. one with metricstransformprocessor and the other without it). So we currently have the flexibility to do it, perhaps it needs to be documented better

Reply:
@pmm-sumo Ha! I totally missed the ability to define several pipelines! 😅

But we still need an option to disable the automatic conversion from Otel to Prometheus/OpenMetrics standard for certain metrics, right?

Example: 

We have a bunch of metrics that are properly converted from Otel to Prometheus standard, except for this one:
*Gauge* `myapp.filesystem.usage_megabytes` with unit `{MBy}` (obviously broken, but it happens)

Automatic conversion would produce `myapp_filesystem_usage_megabytes_MBy` (`MBy` was appended, even though `_megabytes` is already mentioned).

User wants `myapp_filesystem_usage_megabytes` (no `_MBy`).

User will not be able to **metrictransformprocessor** because there is no renaming to be done, and this processor cannot remove the `MBy` unit from metric.

I'm fine simply removing the `rename` map option and even the ability to cover the above example (the code will be just simpler). I just want to be sure I'm not going to be asked to put it back later, ha ha 😉

Reply:
@pmm-sumo @dashpole Should we remove the `namespace` option from the Prometheus exporter? It could be handled with a processor as well. (in a separate PR of course!)

Reply:
I would support that.

Reply:
> Automatic conversion would produce myapp_filesystem_usage_megabytes_MBy (MBy was appended, even though _megabytes is already mentioned).

I think suffixing is actually the correct behavior, given the metric and unit.  Omitting _MBy would result in the output being invalid openmetrics.  It may look better, but we should ensure our output is valid so it isn't rejected by other ecosystem components, which may rely on that behavior.

> But we still need an option to disable the automatic conversion from Otel to Prometheus/OpenMetrics standard for certain metrics, right?

We are discussing that [below](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10028#discussion_r873711876), but i'm not convinced that we need to be able to disable conversion to OM standards.

I would support the ability for a processor to modify metric units to potentially help in cases like this.

Reply:
Okay I will remove these options. Thanks!